app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -59,6 +59,20 @@ def google_oauth2
     elsif google_user
       flash[:success] = I18n.t 'devise.omniauth_callbacks.success', kind: google_site_title
       sign_in_and_redirect google_user, event: :authentication
+    elsif Errbit::Config.google_auto_provision
+      if User.valid_google_domain?(google_email)
+        user = User.create_from_google_oauth2(request.env['omniauth.auth'])
+        if user.persisted?
+          flash[:notice] = I18n.t "devise.omniauth_callbacks.success", kind: google_site_title
+          sign_in_and_redirect user, event: :authentication
+        else
+          session['devise.google_data'] = request.env['omniauth.auth'].except(:extra)
+          redirect_to new_user_session_path, alert: user.errors.full_messages.join("\n")
+        end
+      else
+        flash[:error] = "Account's email domain is not authorized for login. Please log in with an account from a trusted domain."
```

Comment:
Ideally, these messages would use I18n.t so that they could be translated more easily in the future. I can merge without doing that, but it is nice to have.